File: flutter/platform/lib/src/modules/context.dart
import '../core/http_client.dart';
import '../core/api_endpoints.dart';
import '../core/result.dart';

/// Context module — memory recall, store, list, delete, messages.
class ContextModule {
  const ContextModule();

  Future<ApiResult<Map<String, dynamic>>> recall(Map<String, dynamic> body) =>
      HttpClient.instance.postMap(ApiEndpoints.contextRecall, body: body);

  Future<ApiResult<Map<String, dynamic>>> store(Map<String, dynamic> body) =>
      HttpClient.instance.postMap(ApiEndpoints.contextStore, body: body);

  Future<ApiResult<List<dynamic>>> listMemories({
    required String userId,
    String? projectId,
  }) =>
      HttpClient.instance.getRawList(
        ApiEndpoints.contextMemories,
        queryParams: {
          'userId': userId,
          if (projectId != null) 'projectId': projectId,
        },
      );

  Future<ApiResult<void>> deleteMemory(String id, {required String userId}) =>
      HttpClient.instance.deleteVoid(
        ApiEndpoints.contextMemoryDetail(id),
        queryParams: {'userId': userId},
      );

  Future<ApiResult<List<dynamic>>> getMessages({
    required String conversationId,
    String? limit,
  }) =>
      HttpClient.instance.getRawList(
        ApiEndpoints.contextMessages,
        queryParams: {
          'conversationId': conversationId,
          if (limit != null) 'limit': limit,
        },
      );
}

File: js/platform/src/modules/context.ts
import { http } from "../core/http.js";
import { Endpoints } from "../core/endpoints.js";
import type { ApiResult } from "../core/result.js";

export const context = {
  recall(body: {
    userId: string;
    projectId?: string | null;
    query: string;
  }): Promise<ApiResult<Record<string, unknown>>> {
    return http.postMap(Endpoints.contextRecall, body);
  },

  store(body: {
    userId: string;
    projectId?: string | null;
    fact: string;
    category?: string;
    importance?: number;
  }): Promise<ApiResult<Record<string, unknown>>> {
    return http.postMap(Endpoints.contextStore, body);
  },

  listMemories(query: { userId: string; projectId?: string }): Promise<ApiResult<unknown[]>> {
    return http.getList(Endpoints.contextMemories, query);
  },

  deleteMemory(id: string, query: { userId: string }): Promise<ApiResult<null>> {
    return http.deleteVoid(Endpoints.contextMemoryDetail(id), query);
  },

  getMessages(query: { conversationId: string; limit?: string }): Promise<ApiResult<unknown[]>> {
    return http.getList(Endpoints.contextMessages, query);
  },
};
